src/Centrifuge.ts

```diff
@@ -734,22 +734,19 @@ export class Centrifuge {
   _events(chainId: number) {
     return this._query(
       ['events', chainId],
-      () =>
-        using(
-          () => {
-            const subject = new Subject<WatchEventOnLogsParameter>()
-            const unwatch = this.getClient(chainId).watchEvent({
-              onLogs: (logs) => subject.next(logs),
-            })
-            return {
-              unsubscribe: unwatch,
-              subject,
-            }
-          },
-          ({ subject }: any) => subject as Subject<WatchEventOnLogsParameter>
-        ),
-      { cache: false } // Only emit new events
-    ).pipe(filter((logs) => logs.length > 0))
+      () => {
+        return new Observable<WatchEventOnLogsParameter>((subscriber) => {
+          const unwatch = this.getClient(chainId).watchEvent({
+            onLogs: (logs) => subscriber.next(logs),
+          })
+          return unwatch
+        }).pipe(
+          filter((logs) => logs.length > 0),
+          share()
+        )
+      },
+      { cache: true }
```

Comment:
`cache: true` will only cause it to emit past events to new subscribers, which isn't useful to them